v2/yarpcjson/register.go

```diff
@@ -49,23 +49,24 @@ func Procedure(name string, handler interface{}) []yarpc.EncodingProcedure {
 				wrapUnaryHandler(name, handler),
 			),
 			Encoding: Encoding,
-			Codec:    newCodec(name, handler),
+			Codec:    newCodec(handler),
 		},
 	}
 }
 
 // wrapUnaryHandler takes a valid JSON handler function and converts it into a
 // yarpc.UnaryEncodingHandler.
 func wrapUnaryHandler(name string, handler interface{}) yarpc.UnaryEncodingHandler {
+	verifyUnarySignature(name, reflect.TypeOf(handler))
```

Comment:
It looks like we should validate the handler for both `wrapHandler` and `newCodec` calls. Instead of moving it to this function, would you be opposed to moving the `verifyUnarySignature` call to the first call in `Procedure`?

```go
func Procedure(..){
 verifyUnarySignature(name, reflect.TypeOf(handler))
 return []yarpc.EncodingProceure{
   ...
 }
}
```

Reply:
Not at all, I agree that's better.